structs/structs.go

```diff
@@ -0,0 +1,58 @@
+package structs
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// Result represents the comparison outcome of a field, including its name, old value, and new value.
+type Result struct {
+	FieldName string
+	OldValue  interface{}
+	NewValue  interface{}
+}
+
+/*
+CompareStructs compares two struct instances of the same type
+and returns a list of results with the old and new values of each field tagged with `updateable`.
+*/
+func CompareStructs(old, new interface{}) ([]Result, error) {
+	if reflect.TypeOf(old) != reflect.TypeOf(new) {
+		return nil, fmt.Errorf("both structs must be of the same type")
+	}
+
+	oldValue := reflect.ValueOf(old)
+	newValue := reflect.ValueOf(new)
+
+	var comparedResults = make([]Result, 0)
```

Comment:
We should initialize the slice with a capacity, possibly using `make([]Result, 0, oldValue.NumField())`. This avoids unnecessary memory allocations when appending items later, which improves performance.

Reply:
done